### ISSUE TITLE
Java cannot find certification path

### ISSUE THREAD

**Issue by GergelyKalmar** [google-github-actions/auth#396]
### TL;DR

It seems that we get a certification path error when we are using workload identity federation in combination with gcs-connector (see issue https://github.com/GoogleCloudDataproc/hadoop-connectors/issues/1106 and https://github.com/actions/runner-images/issues/9354):

```
Caused by: java.io.IOException: Error getting subject token from metadata server: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
```

For the full stacktrace see https://github.com/logikal-io/mindlab/actions/runs/7940823201/job/21853958926.

### Expected behavior

We were expecting our test suite to succeed when using workload identity federation.

### Observed behavior

Our test suite fails when using workload identity federation. It works when using other auth methods.

### Action YAML

```yaml
See https://github.com/logikal-io/mindlab/blob/update-dependencies/.github/workflows/test-pull-request.yml
```

**Comment by sethvargo** [comment]
Hi @GergelyKalmar - thank you for opening an issue. It looks like the Hadoop connector does not support Workload Identity Federation. If it's using the standard Google Java SDK, it may require a dependency update. Unfortunately there's nothing we can do in this "auth" action to fix that. Upstream clients must add support for Workload Identity.

You could try using the less-secure Service Account Key Export, which will probably work.

**Comment by sethvargo** [comment]
`WORKLOAD_IDENTITY_FEDERATION_CREDENTIAL_CONFIG_FILE` is not a standard envvar, so I'm not sure what's expected by that. The project would need to properly handle `GOOGLE_APPLICATION_CREDENTIALS` pointing to a WIF file (instead of a Service Account Key).

**Comment by elvin-sadigov-db** [comment]
Hi @sethvargo, Sorry I missed above conversation. 
I went through the source code. There is a unit test which reads WIF file, but as you mentioned how the code handle the credential json file, that is the question.
https://github.com/GoogleCloudDataproc/hadoop-connectors/blob/v3.0.0/util-hadoop/src/test/java/com/google/cloud/hadoop/util/HadoopCredentialsConfigurationTest.java#L174

They use below google auth versions and had a release for 3.0.0 version. I believe google auth 1.14.0 supports WIF.
https://github.com/GoogleCloudDataproc/hadoop-connectors/blob/v3.0.0/pom.xml
![image](https://github.com/google-github-actions/auth/assets/105502395/c7881f93-727d-4174-a237-87fa3a9f94c8)


@GergelyKalmar FYI, thanks!
